src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -1972,15 +1976,18 @@ private static string LookupHash(uint h)
             /// that isn't really suspicious.
             /// </summary>
             /// <returns>The string matching the hash, or null.</returns>
-            private static string CheckForMatches(uint[] runningHash, int upTo)
+            private static string CheckForMatches(Span<uint> runningHash, int upTo)
             {
                 var upToMax = runningHash.Length;
                 if (upTo == 0 || upTo > upToMax)
                 {
                     upTo = upToMax;
                 }
 
-                for (var i = 0; i < upTo; i++)
+                runningHash = runningHash.Slice(0, upTo);
+
+                // Starts from where the hash for the shortest suspicious string would be stored.
+                for (var i = MIN_LEN - 1; i < runningHash.Length; i++)
```

Comment:
Given this is not a `for (i = 0; i < span.Length; i++)` pattern, are you sure the boundary check will be eliminated?

Reply:
Yes, this works. https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHiUGAWQAUASg5ceUyADtcGBgEsZCwQEkAcgH0AMgFENDALwMUAbklTuAN2xQGAC2MMZMAO4MwDuwG1yABn8AXQtqKx4AM2gGYVt7RWd1bX1DOAZyMyUGAB5HADodGBkAcwwHTMUAakrRSytOMPCrYnIATmEHH0Ug0VCmhgBfOqHqAaA==

Reply:
That doesn't reflect the actual situation of this method. [This](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8GuGbunHiKBgDkAFAGUADtgB2AHjAALbFAB8DRWgYBLGRgYBXSQBUIASl7d2lvl0gzc+3foCyASWEB9ADIBRYQwAvAwoANw2topBGgB04gA22mAwogAMWkamZuHUttwAZtAMogBuKjrR7l5+AXAM5KEVcrHeMDIA5hiKjdoA1L0WuXkcEcNM5ACcoooA2toAutmj3AC+EWtDXBH0TEIAChLS8koq6po6eoYm5hHWm7b2jheuHj7+0WHLXIVQxWW/2kqrxqDDqDSaV1MPX6gzGdzGfGIk2mc0WOTGGz4GxWQA==) does.

Having `Slice` here moves the boundary check from within the loop to outside the loop, so it is better. Also, it doesn't affect readability, so it's good to have this change.